src/irmin-pack/unix/ext.ml

```diff
@@ -300,6 +300,23 @@ module Maker (Config : Conf.S) = struct
             | From_v3_no_gc_yet -> Ok 0
             | From_v3_gced x -> Ok x.generation
 
+          let unlink_result_file ~root ~generation =
+            let result_file =
+              Irmin_pack.Layout.V3.gc_result ~root ~generation
+            in
+            match Io.classify_path result_file with
+            | `No_such_file_or_directory -> ()
+            | `File -> (
+                match Io.unlink result_file with
+                | Error e ->
+                    [%log.warn
+                      "Unlinking temporary files from previous failed gc. \
+                       Failed with error %a"
+                      Errs.pp e]
+                | Ok () -> ())
+            | `Directory | `Other ->
+                [%log.warn "Unexpected format for %s found" result_file]
+
```

Comment:
You could have called `Io.unlink` straight away and looked at the output to print the right message. It's the _fail-fast principle_. It applies well in this situation.

Anyhow, don't bother modifying your code.

Reply:
I don't mind changing it :)